src/ImageSharp/Processing/Processors/Transforms/Resize/ResizeWorker.cs

```diff
@@ -131,17 +130,18 @@ public void FillDestinationPixels(RowInterval rowInterval, Buffer2D<TPixel> dest
                 ref Vector4 tempRowBase = ref MemoryMarshal.GetReference(tempColSpan);
 
                 int top = kernel.StartIndex - this.currentWindow.Min;
+
                 ref Vector4 fpBase = ref transposedFirstPassBufferSpan[top];
 
-                for (int x = 0; x < this.destWidth; x++)
+                for (nint x = left; x < right; x++)
```

Comment:
Here it's good to have `nint`, as in the `Unsafe.Add`s below this removes the sign extending move `movsxd` which is beneficial.

Reply:
Would it be better to have the loop as
```suggestion
                for (nint x = 0; x < (right - left); x++)
```
so that the subtractions below `x - left` can be eliminated (shifted to the loop variable), so that it's just `x`.
I assume current JIT won't (completely) CSE that expression.

Reply:
Good spot. Not sure how I missed that.